managed/services/agents/registry.go

```diff
@@ -278,15 +288,16 @@ func (r *Registry) unregister(pmmAgentID, disconnectReason string) *pmmAgentInfo
 }
 
 // ping sends Ping message to given Agent, waits for Pong and observes round-trip time and clock drift.
-func (r *Registry) ping(ctx context.Context, agent *pmmAgentInfo) error {
+// Returns true if pong is received, false if there is no pong or error occurred.
+func (r *Registry) ping(ctx context.Context, agent *pmmAgentInfo) (bool, error) {
 	l := logger.Get(ctx)
 	start := time.Now()
 	resp, err := agent.channel.SendAndWaitResponse(&agentv1.Ping{})
 	if err != nil {
-		return err
+		return false, err
 	}
 	if resp == nil {
-		return nil
+		return false, errors.New("pong is not received, response is nil")
```

Comment:
I'm curious what's the reason to introduce boolean variable? it can be simply replaced by comparing if err != nil

Reply:
I forgot that during refactoring I added error to nil response and it is not needed anymore. Thanks for point out.